chapter01/TicTacToe.py

```diff
@@ -246,20 +246,20 @@ def takeAction(self):
 
         values = []
         for hash, pos in zip(nextStates, nextPositions):
-            values.append((self.estimations[hash], pos))
+            values.append((int(self.estimations[hash]), pos))
```

Comment:
`self.estimations[hash]` is a float number within [0, 1], so it should be cast to `int`